### DISCUSSION TITLE
Global variable assignments inconsistent

### DISCUSSION THREAD

**Discussion by zhanghenan-dev** [microsoft/msquic#2718] [Q&A] [answered]
Global variable assignments inconsistent. Please tell me where is wrong?

I add a global variable at src/platform/custom_define.c
volatile uint8_t is_vrrp_enable = 0;

Assignments at tools/sample/sample.c::RunServer()
is_vrrp_enable = 1;

It has compiled properly, start quicsample server, a client has connected, but printf is unexpected.
1. sample.c assignments and printf is expected.
vrrp_enable:1, addr:0x627bf8

2. binding.c printf is unexpected, variable address is different.
QuicBindingReceive vrrp_enable:0, addr:0x7f58ece8a1f8

**Comment by thhous-msft** [comment, marked as answer]
The only APIs exposed are from the function table. Since MsQuic is built into a DLL, global variables will not be exposed outside of the DLL, and assigning in sample.c will not update the variable inside MsQuic itself.

**Reply by zhanghenan-dev** [reply]
> The only APIs exposed are from the function table. Since MsQuic is built into a DLL, global variables will not be exposed outside of the DLL, and assigning in sample.c will not update the variable inside MsQuic itself.

Thanks a lot.
